src/CartingService/Entities/Cart.cs

```diff
@@ -0,0 +1,57 @@
+namespace CartingService.Entities;
+
+using CartingService.Exceptions;
+using CartingService.Interfaces;
+using CartingService.ValueObjects;
+using FluentValidation;
+using OneOf;
+using OneOf.Types;
+
+public class Cart {
+    private readonly ICartRepository _repository;
+    private readonly IValidator<CartItem> _cartItemValidator;
+
+    public Cart(ICartRepository cartRepository, IValidator<CartItem> cartItemValidator) {
+        _repository = cartRepository;
+        _cartItemValidator = cartItemValidator;
+    }
+
+    public async Task<OneOf<List<CartItem>, NotFound>> GetById(CartId id) {
+        var cart = await _repository.GetById(id);
```

Comment:
To my mind this naming is confusing.
Although this implementation meets requirements, it is very hard to follow, as cart repository  method `GetById` returns not a whole cart but only it's items. I'd change methods name to properly reflect what it does or return whole object and then return only object in question.